File: android/app/src/main/java/com/simpleshift/scheduler/domain/model/SalaryBreakdown.kt
package com.simpleshift.scheduler.domain.model

import androidx.compose.runtime.Immutable
import java.time.YearMonth

@Immutable
data class SalaryBreakdown(
    val month: YearMonth,
    val shiftCounts: Map<ShiftType, Int>,
    val shiftPremiumTotal: Double
)

File: flutter/lib/domain/models/salary_breakdown.dart
import 'shift_type.dart';

/// 津贴明细 — 对应 Android 版 SalaryBreakdown.kt
class SalaryBreakdown {
  final int year;
  final int month; // 1..12
  final Map<ShiftType, int> shiftCounts;
  final double shiftPremiumTotal;

  const SalaryBreakdown({
    required this.year,
    required this.month,
    required this.shiftCounts,
    required this.shiftPremiumTotal,
  });
}
